games-rpg/draci-historie/draci-historie-2012-r1.ebuild

```diff
@@ -56,24 +56,28 @@ src_prepare() {
 src_install() {
 	newicon bert.ico draci-historie.ico
 	insinto /usr/share/${PN}
+
+	local lingua
 	for lingua in $(find * -type d); do
 		doins -r ${lingua}
 	done
-	if use l10n_en || ( ! use l10n_cs && ! use l10n_de && ! use l10n_en && ! use l10n_pl ) ; then
-		make_wrapper draci-historie-en "scummvm -f -p \"/usr/share/${PN}/en\" draci" .
-		make_desktop_entry ${PN}-en "Dračí Historie (English)" /usr/share/pixmaps/draci-historie.ico
+
+	if use l10n_en || ( ! use l10n_cs && ! use l10n_de && ! use l10n_en && ! use l10n_pl ); then
+		make_wrapper draci-historie-en "scummvm -f -p \"${EPREFIX}/usr/share/${PN}/en\" draci" .
+		make_desktop_entry ${PN}-en "Dračí Historie (English)" "${EPREFIX}/usr/share/pixmaps/draci-historie.ico"
```

Comment:
Not a full review but this bothers me. Nice to have thought of prefix but this whole thing was wrong to start with, not only should desktop entries not have full path to icons, but icons other than png, xpm, and svg are all invalid as far as XDG is concerned and won't be displayed in most cases.

Optimally the idea would be to convert them and add them to SRC_URI but I've personally been lazy with these and just use the generic "applications-games" icon (i.e. `make_desktop_entry name Name applications-games`) for now and remove bad icons. Going through all of these and eventually giving them proper icons in my devspace is a TODO I'm keeping for later (for old games anyway).

Reply:
Actually, if any icons are bad / missing I'll fix these on merge now -- don't want to get too backlogged on this so started adding them to devspace now.

Reply:
Thanks for doing that. As you suggested i've now simply use the ` applications-games` placeholder (only for `draci-historie` now, but going to update the others too) :)

Reply:
You can add the icon now if you want (similar to the old .ico):
`https://dev.gentoo.org/~ionen/distfiles/${PN}.png` (draci-historie.png)
`doicon "${DISTDIR}"/${PN}.png`
And update Manifest.

For make_desktop_entry, remove the applications-games placeholder it'll use ${PN}.png by default.

In this batch, was there any other missing/wrong? I see xpm and png on others, and that sounds okay (assuming they are square, I've had a case where the icon was 32x33 and it wouldn't display :eyes:).

Reply:
I've checked all icons and the others seems to be fine except for `games-rpg/eschalon-book-1-demo` (which doesn't seems to have a icon). I guess using `applications-games` would be the right way to fix it (or getting a icon for this game) but since it doesn't even start (see comment below) i haven't touched the ebuild for now.